internal/config/config.go

```diff
@@ -638,6 +639,148 @@ func (c *Config) Validate(force bool) error {
+	TEEVersion uint8 `json:"teeVersion" yaml:"teeVersion"`
+	// description: |
+	//   Lowest acceptable SEV-SNP version.
+	SNPVersion uint8 `json:"snpVersion" yaml:"snpVersion"`
+	// description: |
+	//   Lowest acceptable microcode version.
+	MicrocodeVersion uint8 `json:"microcodeVersion" yaml:"microcodeVersion"`
+	// description: |
+	//   Configuration for validating the firmware signature.
+	FirmwareSignerConfig SNPFirmwareSignerConfig `json:"firmwareSignerConfig" yaml:"firmwareSignerConfig"`
+	// description: |
+	//   AMD Root Key certificate used to verify the SEV-SNP certificate chain.
+	AMDRootKey Certificate `json:"amdRootKey" yaml:"amdRootKey"`
+}
+
+// DefaultForAzureSEVSNP returns the default configuration for Azure SEV-SNP attestation.
+// TODO(AB#3042): replace with dynamic lookup for configurable values.
+func DefaultForAzureSEVSNP() AzureSEVSNP {
+	return AzureSEVSNP{
+		Measurements:      measurements.DefaultsFor(cloudprovider.Azure),
+		BootloaderVersion: bootloaderVersion,
+		TEEVersion:        teeVersion,
+		SNPVersion:        snpVersion,
+		MicrocodeVersion:  microcodeVersion,
+		FirmwareSignerConfig: SNPFirmwareSignerConfig{
+			AcceptedKeyDigests: idkeydigest.DefaultList(),
+			EnforcementPolicy:  idkeydigest.MAAFallback,
+		},
+		AMDRootKey: mustParsePEM(arkPEM),
+	}
+}
+
+// GetVariant returns azure-sev-snp as the variant.
+func (AzureSEVSNP) GetVariant() variant.Variant {
+	return variant.AzureSEVSNP{}
+}
+
+// GetMeasurements returns the measurements used for attestation.
+func (c AzureSEVSNP) GetMeasurements() measurements.M {
+	return c.Measurements
+}
+
+// SNPFirmwareSignerConfig is the configuration for validating the firmware signer.
+type SNPFirmwareSignerConfig struct {
+	// description: |
+	//   List of accepted values for the firmware signing key digest.\nValues are enforced according to the 'enforcementPolicy'\n     - 'equal'       : Error if the reported signing key digest does not match any of the values in 'acceptedKeyDigests'\n     - 'maaFallback' : Use 'equal' checking for validation, but fallback to using Microsoft Azure Attestation (MAA) for validation if the reported digest does not match any of the values in 'acceptedKeyDigests'. See the Azure docs for more details: https://learn.microsoft.com/en-us/azure/attestation/overview#amd-sev-snp-attestation\n     - 'warnOnly'    : Same as 'equal', but only prints a warning instead of returning an error if no match is found
+	AcceptedKeyDigests idkeydigest.List `json:"acceptedKeyDigests" yaml:"acceptedKeyDigests"`
+	// description: |
+	//   Key digest enforcement policy. One of {'equal', 'maaFallback', 'warnOnly'}
+	EnforcementPolicy idkeydigest.Enforcement `json:"enforcementPolicy" yaml:"enforcementPolicy" validate:"required"`
+	// description: |
+	//   URL of the Microsoft Azure Attestation (MAA) instance to use for fallback validation. Only used if 'enforcementPolicy' is set to 'maaFallback'.
+	MAAURL string `json:"maaURL,omitempty" yaml:"maaURL,omitempty"`
```

Comment:
the value is ignored if the user sets it in their config file, right?

Reply:
Yes, we only use the value from the ID file.
I could add a check tho to explicitly not allow a set value here

Reply:
You mean something like `validate:"len=0"`?
And regarding the description: Is there any case where a user would see it? For devs, the description is good.

Reply:
The description will only show up if we set a default value (which we dont do)
`validate:"len=0"` sounds like a good idea